### PR TITLE
fix: don't error on already existing dependency

### PR DESCRIPTION
**Pull request by ruben-arts**
This was disscussed in discord, somewhere. But the gist is:

```
pixi add python numpy tensorflow
```
Would fail if any of the packages already "exactly" existed. So by ignoring that error the user is not annoyed by it. The benifit of telling the users they are doing something unneeded didn't out weigh the annoying error.

Before:
```
> pixi add python
Added python
> pixi add python
Duplicate error
```
After:
```
> pixi add python
Added python
> pixi add python
WARN: python already existed
Added python # this is ugly but it needs a bigger refactor to properly solve it.
```
This now also works like this for `pixi remove` where it only warns on removing a non existing package.

### CODE REVIEW

**Comment by olivier-lacroix** [comment]
Hey @ruben-arts , currently, pixi tends to fail for no-ops commands. i.e adding the exact same spec, or removing a package that does not exist in the manifest.

I was a bit surprised by this choice originally, but it was pointed out this was done on purpose, e.g. to avoid ignoring typos. Which made sense.

I feel like we should keep a consistent approach to failing / silently ignoring for commands.

what do you think?

**Comment by ruben-arts** [comment]
Hey @olivier-lacroix, 

Thanks for chiming in! I believe I was strongly for the "error" but the use-case where you would be following documentation/tutorials and then get these errors as well didn't come to mind. This made me want to revisit this change. 

I agree this should be tool wide logic, so users can expect the same behavior everywhere. I currently log the "duplicate" to the DEBUG trace. Do you think that is correct or should it be a little more visible?

And do you know more cli tools that should be changed, except for `add` and `remove`?

**Comment by olivier-lacroix** [comment]
@ruben-arts my pleasure 😄.

I feel having a user-facing Warning, but non-blocking, may be a good middle-ground for those no-ops / not-found / overwriting cases? The debug trace is not going to be seen by users.

Not sure from the top of my head for other CLI commands, but I _think_ the behaviour was fairly consistent for some I looked.

**Comment by tdejager** [comment]
Just for reference, what some package managers seem to do:

Cargo:
```bash
✦ ❯ cargo new --bin nich
    Creating binary (application) `nich` package
note: see more `Cargo.toml` keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
✦ ❯ cd nich/
✦ ❯ l
.rw-r--r--@ 75 tdejager 28 May 13:33 Cargo.toml
drwxr-xr-x@  - tdejager 28 May 13:33 src
✦ ❯ cargo add url
    Updating crates.io index
      Adding url v2.5.0 to dependencies
             Features:
             - debugger_visualizer
             - expose_internals
             - serde
    Updating crates.io index
✦ ❯ cargo add url
    Updating crates.io index
      Adding url v2.5.0 to dependencies
             Features:
             - debugger_visualizer
             - expose_internals
             - serde
✦ ❯ cargo remove url
    Removing url from dependencies
✦ ❯ cargo remove url
    Removing url from dependencies
error: the dependency `url` could not be found in `dependencies`
```

Bun:
```bash
✦ ❯ bun add react
bun add v1.1.10 (5102a944)

installed react@18.3.1

3 packages installed [198.00ms]

✦ ❯ bun add react
bun add v1.1.10 (5102a944)

installed react@18.3.1

[7.00ms] done
✦ ❯ bun remove react
bun remove v1.1.10 (5102a944)
- react
1 package removed [3.00ms]
✦ ❯ bun remove react
bun remove v1.1.10 (5102a944)
```

poetry:
```bash
✦ ❯ poetry add flask
Using version ^3.0.3 for flask

Updating dependencies
Resolving dependencies... (0.4s)

Package operations: 7 installs, 0 updates, 0 removals

  - Installing markupsafe (2.1.5)
  - Installing blinker (1.8.2)
  - Installing click (8.1.7)
  - Installing itsdangerous (2.2.0)
  - Installing jinja2 (3.1.4)
  - Installing werkzeug (3.0.3)
  - Installing flask (3.0.3)

Writing lock file
✦ ❯ poetry add flask
The following packages are already present in the pyproject.toml and will be skipped:

  - flask

If you want to update it to the latest compatible version, you can use `poetry update package`.
If you prefer to upgrade it to the latest available version, you can use `poetry add package@latest`.

Nothing to add.
✦ ❯ poetry remove flask
Updating dependencies
Resolving dependencies... (0.1s)

Package operations: 0 installs, 0 updates, 8 removals

  - Removing blinker (1.8.2)
  - Removing click (8.1.7)
  - Removing colorama (0.4.6)
  - Removing flask (3.0.3)
  - Removing itsdangerous (2.2.0)
  - Removing jinja2 (3.1.4)
  - Removing markupsafe (2.1.5)
  - Removing werkzeug (3.0.3)

Writing lock file

/tmp/dime is 📦 v0.1.0 via 🐍 v3.12.3 on ☁️  tim@prefix.dev took 9s
✦ ❯ poetry remove flask
```

So there is not really a prior art to doing such things:

* Cargo errors on double remove but not on double add.
* Bun does not error for these operations.
* Poetry errors on double adding/removing.

Sorry, this does not really help much, but I thought it might be interesting what other package managers are doing.

**Comment by olivier-lacroix** [comment]
@tdejager this is useful background!

**Comment by tdejager** [comment]
Note that the error code for a double add in poetry is still 0 so it's more of an info message

**Comment by ruben-arts** [comment]
So let's make them not error but warnings right?

**Comment by tdejager** [comment]
> So let's make them not error but warnings right?

Sure, let's do it! We could add a nice message about it already being part of the project, maybe not as silently as bun :)

**Comment by ruben-arts** [comment]
I did it now in a super simple way. resulting in pretty ugly output, but at least it doesn't error out. for both `pixi add` and `pixi remove`

**Review by nichmor — APPROVED** [review]
looks good!

File: src/cli/mod.rs [hunk 1]
```diff
@@ -238,6 +233,7 @@ pub async fn execute() -> miette::Result<()> {
     // Set up the tracing subscriber
     let fmt_layer = tracing_subscriber::fmt::layer()
         .with_ansi(use_colors)
+        .with_target(pixi_level >= LevelFilter::INFO)
```
[the remark this answers is not in the document]
Comment (ruben-arts) [reply src/cli/mod.rs:236]: I was kinda done with the target in the info message so I sneaked it in here. This makes the info messages look a little less daunting to new users.

**Pull request merged**